lib/cocoapods/installer/xcode/pods_project_generator/pod_target_integrator.rb

```diff
@@ -248,6 +249,41 @@ def add_copy_dsyms_script_phase(native_target)
             UserProjectIntegrator::TargetIntegrator.set_input_output_paths(phase, input_paths_by_config, output_paths_by_config)
           end
 
+          # Adds the ODRs that are related to this app spec. This includes the app spec dependencies as well as the ODRs
+          # coming from the app spec itself.
+          #
+          # @param [Xcodeproj::PBXNativeTarget] native_target
+          #         the native target for which to add the ODR file references into.
+          #
+          # @param [Specification] app_spec
+          #         the app spec to integrate ODRs for.
+          #
+          # @return [void]
+          #
+          def add_on_demand_resources(native_target, app_spec)
+            dependent_targets =  target.dependent_targets_for_app_spec(app_spec)
+            parent_odr_group = native_target.project.pod_group(target.name)[app_spec.base_name]
+
+            # Add ODRs of the app spec dependencies first.
+            dependent_targets.each do |pod_target|
+              file_accessors = pod_target.file_accessors.select do |fa|
+                fa.spec.library_specification? ||
+                  fa.spec.test_specification? && pod_target.test_app_hosts_by_spec[fa.spec]&.first == app_spec
```

Comment:
don't know the structure of `pod_target.test_app_hosts_by_spec[fa.spec]` but should this check be `pod_target.test_app_hosts_by_spec[fa.spec]&.include?(app_spec)` rather than only checking `first`?

Reply:
It returns a tuple:
```ruby
    # @return [Hash{Specification => (Specification,PodTarget)}] tuples of app specs and pod targets by test spec.
    #
    attr_accessor :test_app_hosts_by_spec
```

I am interested only on the `first` part to match against the spec.

We could potentially expose a method in `pod_target` but this property is public for now.